packages/components/src/components/hds/table/index.ts

```diff
@@ -4,29 +4,110 @@
+import type { HdsTableThSortArgs } from './th-sort.ts';
+import type { HdsTableTrArgs } from './tr.ts';
+import {
+  HdsTableDensityValues,
+  HdsTableThSortOrderValues,
+  HdsTableVerticalAlignmentValues,
+} from './types.ts';
+import type {
+  HdsTableColumn,
+  HdsTableDensities,
+  HdsTableHorizontalAlignment,
+  HdsTableSelectableRow,
+  HdsTableSortingFunction,
+  HdsTableThSortOrder,
+  HdsTableVerticalAlignment,
+} from './types';
+
+export const DENSITIES: string[] = Object.values(HdsTableDensityValues);
+export const DEFAULT_DENSITY = HdsTableDensityValues.Medium;
+
+export const VALIGNMENTS: string[] = Object.values(
+  HdsTableVerticalAlignmentValues
+);
+export const DEFAULT_VALIGN = HdsTableVerticalAlignmentValues.Top;
+
+export interface HdsTableArgs {
+  Args: {
+    align?: HdsTableHorizontalAlignment;
+    caption?: string;
+    columns?: HdsTableColumn[];
+    density?: HdsTableDensities;
+    identityKey?: string;
+    isFixedLayout?: boolean;
+    isSelectable?: boolean;
+    isStriped?: boolean;
+    model: Record<string, unknown>;
+    onSelectionChange?: (selection: {
+      selectionKey: string;
+      selectionCheckboxElement:
+        | HdsFormCheckboxBaseSignature['Element']
+        | undefined;
+      selectedRowsKeys: string[];
+      selectableRowsStates: {
+        selectionKey: string;
+        isSelected: boolean | undefined;
+      }[];
+    }) => void;
+    onSort?: (sortBy: string, sortOrder: HdsTableThSortOrder) => void;
+    selectionAriaLabelSuffix?: string;
+    sortBy?: string;
+    sortedMessageText?: string;
+    sortOrder?: HdsTableThSortOrder;
+    valign?: HdsTableVerticalAlignment;
+  };
+  Blocks: {
+    head?: [
+      {
+        Tr: ComponentLike<HdsTableTrArgs>;
+        Th: ComponentLike<HdsTableThArgs>;
+        ThSort: ComponentLike<HdsTableThSortArgs>;
+        sortBy: string | undefined;
+        sortOrder: HdsTableThSortOrder;
+        setSortBy: (column: string) => void;
+      },
+    ];
+    body?: [
+      {
+        Td: ComponentLike<HdsTableTdArgs>;
+        Tr: ComponentLike<HdsTableTrArgs>;
+        Th: ComponentLike<HdsTableThArgs>;
+        ThSort?: ComponentLike<HdsTableThSortArgs>;
+        data?: string;
+        sortBy?: string | undefined;
+        sortOrder?: HdsTableThSortOrder;
+      },
+    ];
+  };
+  Element: HTMLTableElement;
+}
 
-export const DENSITIES = ['short', 'medium', 'tall'];
-const DEFAULT_DENSITY = 'medium';
-const VALIGNMENTS = ['top', 'middle', 'baseline'];
-const DEFAULT_VALIGN = 'top';
-
-export default class HdsTableIndexComponent extends Component {
+export default class HdsTableIndexComponent extends Component<HdsTableArgs> {
   @tracked sortBy = this.args.sortBy;
-  @tracked sortOrder = this.args.sortOrder || 'asc';
-  @tracked selectAllCheckbox = undefined;
-  selectableRows = [];
-  @tracked isSelectAllCheckboxSelected = undefined;
+  @tracked sortOrder = this.args.sortOrder || HdsTableThSortOrderValues.Asc;
```

Comment:
```suggestion
  @tracked sortOrder = this.args.sortOrder ?? HdsTableThSortOrderValues.Asc;
```

Reply:
Since it is possible any of our consumers are passing in an empty string for @sortOrder I am going to leave current code in place and not do any refactors that could possibly break something.

- **Logical OR (`||`)**: Defaults to the right-hand side value if the left-hand side value is any falsy value (e.g., `false`, `0`, `''`, `null`, `undefined`).

- **Nullish Coalescing (`??`)**: Defaults to the right-hand side value only if the left-hand side value is `null` or `undefined`.

### Example
```typescript
@tracked sortOrder = this.args.sortOrder || HdsTableThSortOrderValues.Asc;
// Uses default if this.args.sortOrder is falsy (false, 0, '', null, undefined)

@tracked sortOrder = this.args.sortOrder ?? HdsTableThSortOrderValues.Asc;
// Uses default only if this.args.sortOrder is null or undefined
```

Use `||` for broader defaulting and `??` for more precise defaulting.